src/main/java/tdl/s3/helpers/MD5Digest.java

```diff
@@ -8,15 +8,15 @@
 
     private static MessageDigest md5Digest;
 
-    static {
-        try {
-            md5Digest = MessageDigest.getInstance("MD5");
-        } catch (NoSuchAlgorithmException e) {
-            throw new RuntimeException("Can't send multipart upload. Can't create MD5 digest. " + e.getMessage(), e);
-        }
-    }
-
     public static String digest(byte[] bytes) {
```

Comment:
Instead of using PowerMock, we should make the algorithm a parameter to the function and have the clients specify with algorithm they need. The class name might have to change too.